heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -0,0 +1,225 @@
+          this._eslintToolPath,
+          this._eslintConfigFilePath,
+          tsProgram,
+          changedFiles
+        )
+      );
+    }
+    if (this._tslintToolPath && this._tslintConfigFilePath) {
+      lintingPromises.push(
+        this._runTslintAsync(
+          taskSession,
+          heftConfiguration,
+          this._tslintToolPath,
+          this._tslintConfigFilePath,
+          tsProgram,
+          changedFiles
+        )
+      );
+    }
+
+    return await Promise.all(lintingPromises);
+  }
+
+  private async _runEslintAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    eslintToolPath: string,
+    eslintConfigFilePath: string,
+    tsProgram: IExtendedProgram,
+    changedFiles?: Set<IExtendedSourceFile> | undefined
+  ): Promise<Eslint> {
+    const eslint: Eslint = await Eslint.loadAsync({
+      ts: this._typeScript,
+      scopedLogger: taskSession.logger,
+      eslintPackagePath: eslintToolPath,
+      linterConfigFilePath: eslintConfigFilePath,
+      buildFolderPath: heftConfiguration.buildFolder,
+      buildMetadataFolderPath: taskSession.cacheFolder
+    });
+
+    eslint.printVersionHeader();
+
+    const typeScriptFilenames: Set<string> = new Set(tsProgram.getRootFileNames());
+    await eslint.performLintingAsync({
+      tsProgram,
+      typeScriptFilenames,
+      changedFiles: changedFiles || new Set(tsProgram.getSourceFiles())
+    });
+
+    return eslint;
+  }
+
+  private async _runTslintAsync(
+    taskSession: IHeftTaskSession,
+    heftConfiguration: HeftConfiguration,
+    tslintToolPath: string,
+    tslintConfigFilePath: string,
+    tsProgram: IExtendedProgram,
+    changedFiles?: Set<IExtendedSourceFile> | undefined
+  ): Promise<Tslint> {
+    const tslint: Tslint = await Tslint.loadAsync({
+      ts: this._typeScript,
+      scopedLogger: taskSession.logger,
+      tslintPackagePath: tslintToolPath,
+      linterConfigFilePath: tslintConfigFilePath,
+      buildFolderPath: heftConfiguration.buildFolder,
+      buildMetadataFolderPath: taskSession.cacheFolder
+    });
+
+    tslint.printVersionHeader();
+
+    const typeScriptFilenames: Set<string> = new Set(tsProgram.getRootFileNames());
+    await tslint.performLintingAsync({
+      tsProgram,
+      typeScriptFilenames,
+      changedFiles: changedFiles || new Set(tsProgram.getSourceFiles())
+    });
+
+    return tslint;
+  }
+
+  private async _resolveEslintConfigFilePathAsync(heftConfiguration: HeftConfiguration): Promise<string> {
+    // When project is configured with "type": "module" in package.json, the config file must have a .cjs extension
+    // so use it if it exists
+    const defaultPath: string = path.resolve(heftConfiguration.buildFolder, '.eslintrc.js');
+    const alternativePathPath: string = path.resolve(heftConfiguration.buildFolder, '.eslintrc.cjs');
+    return await FileSystem.existsAsync(alternativePathPath) ? alternativePathPath : defaultPath;
```

Comment:
This feels a little janky. What if both exist? What about:


```suggestion
    const [alternativePathExists, defaultPathExists] = await Promise.all([
      FileSystem.existsAsync(alternativePath),
      FileSystem.existsAsync(defaultPath)
    ]);
    if (alternativePathExists && defaultPathExists) {
      throw new Error(`This project contains both a ${ESLINTRC_JS_FILENAME} and a ${ESLINTRC_CJS_FILENAME}. Only one may be present.`); // But probably with better phrasing.
    } else if (alternativePathExists) {
      return alternativePath;
    } else if (defaultPathExists) {
      return defaultPath;
    } else {
      return undefined;
    }
```